lib/jsdom/browser/Window.js

```diff
@@ -60,15 +57,27 @@ function Window(options) {
   ///// INTERFACES FROM THE DOM
   // TODO: consider a mode of some sort where these are not shared between all DOM instances
   // It'd be very memory-expensive in most cases, though.
-  for (const name in dom) {
+  this._core = bootstrapDOM("Window", window, { window });
+  for (const name in extraDOM) {
     Object.defineProperty(window, name, {
       enumerable: false,
       configurable: true,
       writable: true,
-      value: dom[name]
+      value: extraDOM[name]
     });
+
+    this._core[name] = extraDOM[name];
   }
-  this._core = dom;
+  // TODO move this somewhere else.
```

Comment:
Running this on every window initialization seems sad since it's all a shared object.